[FIX] themeSwitch class + [ADD] password component + ...

hamid-gh98 opened the pull request:
- fix darkClass in themeSwitcher
- update sub remarks
- add password component
- update docker-compose
- fix and update some translations

alireza0 commented on web/assets/js/util/common.js:
```diff
@@ -69,48 +69,39 @@ function debounce(fn, delay) {
+            return "red";
     }
 }
-
-function areAllItemsExist(array1, array2) {
-    for (let i = 0; i < array1.length; i++) {
-      if (!array2.includes(array1[i])) {
-        return false;
-      }
-    }
-    return true;
-  }
```
It is about a conflict. Please fix it

alireza0 commented on web/html/xui/form/tls_settings.html:
This one is updated in parallel. Please rebase it.

alireza0 reviewed (changes requested):
Good job!
Please fix this minor changes due happened to parallel commits.

hamid-gh98 commented:
Hello @alireza0,
Done.

hamid-gh98 commented:
btw, when I go to settings.html, it says that `areAllItemsExist` is not defined.
so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.

![image](https://github.com/alireza0/x-ui/assets/70919649/31d46c45-7c22-471a-b202-003d66a6035f)

hamid-gh98 commented:
I also now split the country configs (block + direct) in settings page.
hope that helps better

alireza0 commented:
> btw, when I go to settings.html, it says that `areAllItemsExist` is not defined. so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.
> 
> ![image](https://user-images.githubusercontent.com/70919649/238482339-31d46c45-7c22-471a-b202-003d66a6035f.png)

Actually it was defined [here](https://github.com/alireza0/x-ui/blob/main/web/assets/js/util/common.js#L109:L116) , but it seems in your branch it was not updated.
It is better to have general functions in `common.js` instead of directly push it to busy file like settings.
Please move `areAllItemsExist` or `doAllItemsExist` to its right place.
Overall your job is great! Thanks a lot

hamid-gh98 commented:
> > btw, when I go to settings.html, it says that `areAllItemsExist` is not defined. so, I moved its function to `settings.html` and renamed it to `doAllItemsExist`.
> 
> Actually it was defined [here](https://github.com/alireza0/x-ui/blob/main/web/assets/js/util/common.js#L109:L116) , but it seems in your branch it was not updated. It is better to have general functions in `common.js` instead of directly push it to busy file like settings. Please move `areAllItemsExist` or `doAllItemsExist` to its right place. Overall your job is great! Thanks a lot

ok, i removed it from `settings.html`.
thanks.

alireza0 commented:
Well done!
